src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -616,14 +616,15 @@ public static function getTrustedHosts()
     /**
      * Normalizes a query string.
      *
-     * It builds a normalized query string, where keys/value pairs are alphabetized,
+     * It builds a normalized query string, where keys/value pairs are alphabetized (by default),
      * have consistent escaping and unneeded delimiters are removed.
      *
-     * @param string $qs Query string
+     * @param string $qs            Query string
+     * @param bool   $preserveOrder Flag to preserve the querystring from being alphabetically ordered
      *
      * @return string A normalized query string for the Request
      */
-    public static function normalizeQueryString($qs)
+    public static function normalizeQueryString($qs, bool $preserveOrder = false)
```

Comment:
Adding a new argument is a BC break, isn't it?
This needs a BC layer.

Reply:
Totally right, I'll work on this sorry.